lib/loadConfig.js

```diff
@@ -56,14 +57,30 @@ const resolveConfig = (configPath) => {
 
 /**
  * @param {string} [configPath]
+ * @param {Logger} [logger]
  */
-export const loadConfig = async (configPath) => {
-  const explorer = lilconfig('lint-staged', { searchPlaces, loaders })
-  const result = await (configPath ? explorer.load(resolveConfig(configPath)) : explorer.search())
-  if (!result) return null
+export const loadConfig = async (configPath, logger) => {
+  try {
+    if (configPath) {
+      debugLog('Loading configuration from `%s`...', configPath)
+    } else {
+      debugLog('Searching for configuration...')
+    }
 
-  const { config, filepath } = result
+    const explorer = lilconfig('lint-staged', { searchPlaces, loaders })
 
-  // config is a promise when using the `dynamicImport` loader
-  return { config: await config, filepath }
+    const result = await (configPath ? explorer.load(resolveConfig(configPath)) : explorer.search())
+    if (!result) return null
```

Comment:
Maybe also log something in this case?

Reply:
This will get logged here:

https://github.com/okonet/lint-staged/blob/e96b6d9674f07b4686876cb40605274577925973/lib/index.js#L67-L70